# SNP_match_sample
Codes for scRNA sample demultiplexing using Souporcell or Demuxlet

To match the scRNA-seq data for each cell to its donor, we made use of the position-sorted bam files as outputs from Cellranger pipeline v.7.1.0 and the vcf files generated from WGS data (fastq) using nf-core Sarek pipeline v.3.1.2 (GRCh38, haplotypecaller, default parameters). The vcf files were filtered, sorted and merged according to the scRNA-seq samples [(PreprocessVCF.sh)](./PreprocessVCF.sh). Then, the sorted bam files and the processed vcf files were subjected to [Souporcell v.2.0](https://pubmed.ncbi.nlm.nih.gov/32366989/) and [Demuxlet v.1.0](https://pubmed.ncbi.nlm.nih.gov/29227470/) for clustering and assignment. For Souporcell, the filtered matrices used for scRNA-seq analysis were also provided to exclude unused data, and the output clusters were matched to sample id (Match_sampleid_souporcell.R)[./Match_sampleid_souporcell.R]. For Demuxlet, the results using genotype (‘field=GT’) for calculation slightly differed (~10%) from those using Phred-scaled genotype likelihoods (‘field=PL’). The scripts were provided as: [Souporcell.sh](./Souporcell.sh); [Demuxlet.sh](./Demuxlet.sh).

We also compared the results from these two pipelines [(Compare_methods.R)](./Compare_methods.R). For Souporcell, all cells provided in the filtered matrices were assigned to the most likely sample. For Demuxlet, the cells with their identity assigned by both Souporcell and Demuxlet (‘assigned’) and those only in the output of Souporcell (‘dropouts’) were recorded.

